packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -356,5 +388,54 @@ private MethodBodyStatement GetPropertyInitializers(
 
             return rawDataField;
         }
+
+        private PropertyProvider? BuildAdditionalProperties()
+        {
+            var additionalProperties = _inputModel.AdditionalProperties;
+            if (additionalProperties is null)
+            {
+                return null;
+            }
+
+            var valueType = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(additionalProperties);
+            if (valueType is null)
+                throw new InvalidOperationException($"Failed to create CSharpType for additional properties of model {_inputModel.Name}");
+
+            var originalType = new CSharpType(typeof(IDictionary<,>), typeof(string), valueType);
+            var additionalPropsType = !_inputModel.Usage.HasFlag(InputModelTypeUsage.Input)
+                ? ReplaceUnverifiableType(originalType).OutputType
+                : ReplaceUnverifiableType(originalType);
+
+            return new PropertyProvider(
+                null,
+                MethodSignatureModifiers.Public,
+                additionalPropsType,
+                name: AdditionalPropertiesPropertyName,
+                new AutoPropertyBody(false),
+                enclosingType: this);
+        }
+
+        private CSharpType ReplaceUnverifiableType(CSharpType type)
+        {
+            return type switch
+            {
+                _ when type.Equals(_additionalPropsUnknownType, ignoreNullable: true) => type,
+                _ when type.IsFrameworkType && _verifiableAdditionalPropertyTypes.Contains(type.FrameworkType) => type,
+                _ when type.IsUnion => type,
+                _ when type.IsList => type.MakeGenericType([ReplaceUnverifiableType(type.Arguments[0])]),
+                _ when type.IsDictionary => type.MakeGenericType([ReplaceUnverifiableType(type.Arguments[0]), ReplaceUnverifiableType(type.Arguments[1])]),
+                _ => CSharpType.FromUnion([type])
+            };
+        }
+
+        private static readonly HashSet<Type> _verifiableAdditionalPropertyTypes =
```

Comment:
What does verifiable mean here?

Reply:
This is another change I pulled in from [autorest](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Builders/BuilderHelpers.cs#L427). These are the known value types for additional properties that have an additional value kind check when deserializing. You can see an example of what the generated code would look like for an additional properties property whose value type falls in this category [here](https://github.com/Azure/autorest.csharp/blob/67250bdd71c48b6214f590986c83ee849ba54511/test/CadlRanchProjectsNonAzure/type/property/additional-properties/src/Generated/Models/DifferentSpreadStringRecord.Serialization.cs#L81).